### DISCUSSION TITLE
Find Element WITHIN Element doesn't work

### DISCUSSION THREAD

**Discussion by Vhyseris** [ElSnoMan/pyleniumio#342] [Q&A] [answered]
As I read the documentation it should be possible to search within an element, so you don't search the entire dom...
But somehow it doesn't work for me!

```
row = py.getx("//div//table//tr")
cbbyrow = row.getx("//input")
```

Did I miss something, can somebody help?

**Comment by Vhyseris** [comment]
@ElSnoMan ?

**Comment by ElSnoMan** [comment]
What are the errors you're seeing? The more details you can provide, the better

**Reply by Vhyseris** [reply]
Thanks for the reply!

There is no error.
The problem is, that this code returns the first input of the dom, which is outside of the row and the table in general.
It should return me the first input inside of the row, right?

**Comment by ElSnoMan** [comment]
In theory, yes, but I need more context. Can you provide the website or
HTML that you're querying? The xpath you're using doesn't look "good" to
me, so there may be a better solution

**Comment by Vhyseris** [comment]
It's just an simple example html, to test the search within elements...
```
<body>
    <h1>Input and Table Example</h1>

    <!-- Input Field -->
    <input type="text" placeholder="Type something here..."><br>

    <!-- Table -->
    <div>
        <table>
            <thead>
                <tr>
                    <th>
                        <input type="text" placeholder="Type something here...">
                    </th>
                    <th>Age</th>
                    <th>City</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td>Alice</td>
                    <td>25</td>
                    <td>Wonderland</td>
                </tr>
                <tr>
                    <td>Bob</td>
                    <td>30</td>
                    <td>San Francisco</td>
                </tr>
                <tr>
                    <td>Charlie</td>
                    <td>35</td>
                    <td>New York</td>
                </tr>
            </tbody>
        </table>
    </div>
</body>
```

The thing is, that i will not have ids for html elements and have to search over not ideal xpaths, because websites could be generated dynammically...  
Therefore I really need the search within elements to work, to make it easier for me in future.

**Comment by Vhyseris** [comment, marked as answer]
Okayyy, thanks to ChatGPT and the right question I found out what the problem was.

Its not a problem with the pylenium code, but you need to know if your searching within an element **over** XPath, you need to **add a dot** at the beginning! Here, the dot (.) ensures the XPath starts from the parent element instead of the root of the DOM. This approach ensures scoped searches, avoiding performance hits and ensuring the intended element is found more accurately.

**Corrected code**
```
row = py.getx("//div//table//tr")
# Wrong way
# cbbyrow = row.getx("//input")

# Correct way
cbbyrow = row.getx(".//input")
```

May I suggest to add this info to this section of the documentation:  
https://docs.pylenium.io/element-commands/find-elements  

@ElSnoMan 
Thank you for replying and the input!
